Core/main.cpp

```diff
@@ -1,8 +1,142 @@
-#include <iostream>
-
 #include <glad/glad.h>
 #include <GLFW/glfw3.h>
 
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <sstream>
+
+#include "primitives.h"
+
+#define ASSERT(x) if (!(x)) __debugbreak();
```

Comment:
All these OpenGL functions are duplicated code from renderer.cpp, no duplicated code in repo please.

This should also not be in main.cpp, see Architecture's latest expectcations for structure

Reply:
Left this until main successfully creates a rendering object which can draw primitives. Duplication will be deleted on next push.